Felipe - felipegermanos/Week 10/fgc-week_10.do

```diff
@@ -0,0 +1,191 @@
+global tz_elec_10_clean "$user/Econ490_Fall2022/week_10/02_data/Tz_elec_10_clean.dta"
+global tz_15_10_gis "$user/Econ490_Fall2022/week_10/02_data/Tz_GIS_2015_2010_intersection.dta"
+
+
+
+cd "$dir"
+
+********************************************************************************
+******************************   Question 1   **********************************
+********************************************************************************
+
+/* Just a slightly more complete version than the code we used in class */
+
+
+
+import excel "$civ_density", sheet("Population density") firstrow clear
+
+/* First, we drop all rows that don't contain department level data and
+ clean the department names */
+
+keep if regex(NOMCIRCONSCRIPTION, "DEPARTEMENT")
+
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, "DEPARTEMENT", "",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " DE ", "",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " DU ", "",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " DES ", "",.)
+replace NOMCIRCONSCRIPTION =  subinstr(NOMCIRCONSCRIPTION,"D' ","",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION, " D'", "",.)
+
+
+replace NOMCIRCONSCRIPTION = lower(strtrim(strtrim(NOMCIRCONSCRIPTION)))
+gen department = NOMCIRCONSCRIPTION
+drop NOMCIRCONSCRIPTION
+
+/* Looking at the data, it is possible to find one typo in the department names */
+replace department="arrha" if department=="arrah"
+
+
+/* Finally, we save the temporary file*/
+tempfile density
+save `density'
+
+
+/* Here is the code we use to merge the datasets on "department" */
+use "$civ_section0", clear 
+decode b06_departemen, generate(department)
+merge m:1 department using `density'
+
+
+/* Finally, we drop empty rows (case when merge ==2) */
+drop if _merge==2 
+drop _merge
+
+
+/* And this exports the dataset */
+save question_1, replace
+
+
+
+
+
+********************************************************************************
+******************************   Question 2   **********************************
+********************************************************************************
+
+/* We also covered most of this algorythm in class - ordering latitude and longitude
+and choosing appropriate clusters following such ordering. Notice that we need to 
+use a package, geodist, to run this code, so there might be an error because of it */
+
+
+use "$gps_data", clear
+
+
+/* We start by generating a scatterplot to check for clusters of observations */
+sort latitude longitude
+scatter longitude latitude, title("Q2 - Clusters by latitude and longitude")
+
+/* Next, we divide the data, which is sorted, in blocks of six, to reflect
+the fact that each field worker is assigned to 6 households, and use geodist to 
+check how efficient our ordering was. Notice nothing in the code is specific to 
+this dataset, so it should be generalizible */
+egen field_worker = seq(), block(6) 

```

Comment:
This is a decent solution, and I like that it is generalizable. One way to improve it would be assigning enumerators in a way that isn't just based on sorting by latitude and longitude. An issue with this is that, if two houses have very similar latitude but much different longitude, you may assign an enumerator to cover both households despite the two being on quite far apart. For example, look at field_worker 8 in your dataset. They travel from one of the furthest west points to the furthest east point in the dataset.

I would recommend using the "cross" command to form pairwise combinations of all the different points and then using it to calculate the distance between any two locations. Then you can assign enumerators by picking one point and assigning them to that point and the next closest 5 households, removing those households from the dataset, and repeating. Feel free to look at my submission for more details if you're interested, but my method isn't perfect.